student_and_teacher.rb

```diff
@@ -0,0 +1,47 @@
+class Person
+  attr_reader :id
+
+  attr_accessor :age, :name
+
+  def initialize(age, name = 'unknown', parent_permission: true)
```

Comment:
Hi, You have done a great job of creating the Person classe and implementing the required methods. However, there are some issues that need to be fixed or improved. Here are my suggestions:

- You have not assigned an @id instance variable to the Person objects. This is required by the project specifications. You can use a class variable @@id_count to keep track of the number of Person objects created and assign a unique id to each object in the constructor.

Please make these changes and resubmit your code for another review. If you have any questions, feel free to ask. Keep up the good work!